pkg/internal/common/jsoncanonicalizer/jsoncanonicalizer.go

```diff
@@ -255,20 +256,35 @@ func Transform(jsonData []byte) (result []byte, e error) {
 	}
 
 	parseArray = func() string {
-		var arrayData strings.Builder
-		arrayData.WriteByte('[')
+		var elements []string
+
 		var next bool = false
 		for globalError == nil && testNextNonWhiteSpaceChar() != ']' {
 			if next {
 				scanFor(',')
-				arrayData.WriteByte(',')
 			} else {
 				next = true
 			}
-			arrayData.WriteString(parseElement())
+			elements = append(elements, parseElement())
 		}
 		scan()
-		arrayData.WriteByte(']')
+
+		// sort the elements so we enforce an order
```

Comment:
I would also move our changes to a new file so that we have pretty clear separation between the original and our additions.